gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala

```diff
@@ -194,7 +194,15 @@ object GlutenWriterColumnarRules {
                       aqe.isSubquery,
                       supportsColumnar = true
                     ))))
-            case other => rc.withNewChildren(Array(FakeRowAdaptor(other)))
+            case other =>
+              if (other.supportsColumnar) {
+                rc.withNewChildren(Array(FakeRowAdaptor(other)))
+              } else {
+                rc.withNewChildren(
+                  Array(
+                    FakeRowAdaptor(
+                      BackendsApiManager.getSparkPlanExecApiInstance.genRowToColumnarExec(other))))
```

Comment:
Thanks for your PR!
I note `doExecuteColumnar` at `FakeRowAdaptor` has a similar handling for row-based child operator. Doesn't that piece of code work?
If there is any issue found in your test, could you please add some UTs? Thanks!